### DISCUSSION TITLE
Rethink development process

### DISCUSSION THREAD

**Discussion by mcrot** [ContactEngineering/topobank#942] [General]
So far we are following here the ["git flow" workflow](https://nvie.com/posts/a-successful-git-branching-model/) which is still great for software which is deployed in multiple places, e.g. installed on many machines. It is centered around the idea of an "release".

For web-based applications often a simpler approach is applied: [GitHub Flow](https://githubflow.github.io/)
Here, there is no *develop* branch, feature branches are created from the *master* branch and pushes to *master* are 
pushed to the production instance without waiting for new release.

Maybe this is interesting for the development of the topobank main package.

**Comment by pastewka** [comment]
I agree that this would be good. However, we need to think how to

1) handle plugin dependencies (do they still work with current master?)
2) actually implement the automatic deploy

The solution to 1) is probably not to deploy from _topobank_ master but from some production stack repo that contains _topobank_ and plugins as a submodule.

**Comment by Salman-dott** [comment]
Having read the full topic, I still haven't found the answer to my question, unfortunately. I read that blockchain development services [Dysnix.com](https://dysnix.com/) will provide you with more progressive and innovative solution for software development. This is yet to be studied more thoroughly by me, but happy customers' reviews are what make me feel optimistic about it.
